### PR TITLE
some fixes and code quality improvements

### PR DESCRIPTION
**Pull request by hankem**
* The menu state was not always updated correctly when trying to change the selection with the arrow keys.
* The request JSON document is now built using `jq` to supports user input that needs to be quoted.
* [ShellCheck](https://www.shellcheck.net/) is introduced as test workflow. Existing warnings have been fixed or ignored.
* Some formatting improvements for `README.md`.

The changes have been tested on `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`. 
I hope that they're sufficiently compatible, but testing them with other versions might be advisable, if available.

Please let me know if you'd prefer smaller PRs for the individual changes.

### CODE REVIEW

File: please.sh [hunk 1]
```diff
@@ -96,56 +92,55 @@ display_help() {
...
-  response=(${response[@]})
   httpStatus="${response[${#response[@]}-1]}"
-  result=${response[@]::${#response[@]}-1}
 
   if [ "${httpStatus}" -ne 200 ]; then
     >&2 echo "Error: Received HTTP status ${httpStatus}"
     exit 1
   else
-    message=$(echo "${result}" | jq '.choices[0].message.content' --raw-output)
+    message=$(echo "${response[0]}" | jq '.choices[0].message.content' --raw-output)
```
Comment (hankem) [inline please.sh:129]: As far as I understand, `result` should have been an array of all entries of the `response` array except for the last one, but `"${result}"` would only have considered the first element of `result`.

File: please.sh [hunk 2]
```diff
@@ -96,56 +92,55 @@ display_help() {
 
 get_command() {
   role="You translate the input given into Linux command. You may not use natural language, but only a Linux commands as answer."
-  prompt="${commandDescription}"
 
-  payload="{
-    \"model\": \"${model}\",
-    \"messages\": [{\"role\": \"system\", \"content\": \"${role}\"}, {\"role\": \"user\", \"content\": \"${prompt}\"}]
-  }"
+  payload=$(printf %s "$commandDescription" | jq --slurp --raw-input --compact-output '{
```
Comment (hankem) [inline please.sh:96]: Tested with `jq-1.6`. I hope it's sufficiently compatible! 🤞

File: please.sh [hunk 3]
```diff
@@ -96,56 +92,55 @@ display_help() {
...
+    model: "'"$model"'",
+    messages: [{ role: "user", content: . }]
+  }')
 
   explanation=$(perform_openai_request)
 }
 
 perform_openai_request() {
-  response=$(curl https://api.openai.com/v1/chat/completions \
+  IFS=$'\n' read -r -d '' -a response < <(curl https://api.openai.com/v1/chat/completions \
```
Comment (hankem) [inline please.sh:117]: Tested with `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`. I hope it's sufficiently compatible! 🤞

**Comment by thomas-endres-tng** [comment]
Will try it on Mac as well. Thanks for all the improvements.

**Pull request merged**